File: Ejercicio4.hs
-- Verificador par o impar - Haskell

-- Función pura que devuelve un String
paridad :: Int -> String
paridad n
  | even n    = show n ++ " es par."
  | otherwise = show n ++ " es impar."

-- Ejemplo de main para uso por consola (opcional)
main :: IO ()
main = do
  putStrLn "Ingresa un número entero:"
  linea <- getLine
  case reads linea :: [(Int, String)] of
    [(n, "")] -> putStrLn (paridad n)
    _         -> putStrLn "Entrada inválida: por favor ingresa un número entero."

File: Ejercicio4.py
# Verificador par o impar - Python

def es_par(n):
    return n % 2 == 0

def main():
    entrada = input("Ingresa un número entero: ").strip()
    try:
        n = int(entrada)
    except ValueError:
        print("Entrada inválida: por favor ingresa un número entero.")
        return

    if es_par(n):
        print(f"{n} es par.")
    else:
        print(f"{n} es impar.")

if __name__ == "__main__":
    main()
